src/main/java/com/list/LinkedList.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * File: LinkedList.java
+ * Author: Tarun Jain
+ * Date: 1 Feb, 2016
+ * -----------------------------------------
+ * LinkedList is a generic data structure to store data items
+ * of any type in list fashion.  
+ * 
+ *  
+ */
+package com.list;
+
+/**
+ * LinkedList is a generic data structure to store elements in 
+ * list manner.
+ *
+ * @author Tarun Jain
+ * @version 1.0
+ * @param <Item>
+ * 
+ */
+
+public class LinkedList<Item> {
+
+	private Node head;
+	private int listCount;
+	
+	/**
+	 * creates a new empty 
+	 * LinkedList object.
+	 */
+	public LinkedList() {
+		head = new Node();
```

Comment:
Whenever you instantiate a link list, it has to be empty. Do not assign a new node to it. Write:
`
head = null;
N = 0;
 `